tstore/tsdf/ts_class.py

```diff
@@ -116,5 +116,7 @@ def to_disk(self, fpath, partitioning_str=None):
 
     def __repr__(self):
         """Print TS object."""
-        # return self.data.__repr__()
-        return f"TS[shape={self.data.shape},start={self.data.index.min()},end={self.data.index.max()}]"
+        try:
+            return f"TS[shape={self.data.shape},start={self.data.index.min()},end={self.data.index.max()}]"
```

Comment:
This does not work for dask data (the only implemented now). The __repr__ depends on the backend, I see two potential approaches:
- subclassing `TS` depending on the backend and implementing a specific __repr__ for each subclass
- creating the shape, start and end properties (or a single "_repr" property) that are computed once using an `if isinstance(self.data, pd.DataFrame):` and cached (I suppose you understand what I mean but just in case, example here: https://github.com/martibosch/pylandstats/blob/main/pylandstats/landscape.py#L567-L576).

What do you think it is best?

In any case, this is tricky for dask:
- shape: getting the number of columns is straightforward, no issues here, but i have only found one way of getting the number of rows without `.compute()` (see https://stackoverflow.com/a/56835689/17990860), but maybe in our case since we are certain to be loading from a TStore we can make some assumptions and therefore use something like `data.index.size.compute()`
- start/end: again, since we assume that we are loading from a TStore, we can use the partitioning to get only the first and last parquet files, but we'd still need to read them and get the first and last rows respectively (also assuming each parquet file is sorted, which I do not think we enforce when writing to a tstore - am I wrong?).

Altogether, I wonder if we should use a simpler (less informative) `__repr__`, at least for dask data.